app/controllers/application_controller.rb

```diff
@@ -61,6 +61,14 @@ def render_500(e = nil)
     render template: 'errors/error_500', status: 500, layout: 'application', content_type: 'text/html'
   end
 
+  # カンファレンス開催前、かつCFP中
+  # カンファレンス開催前、かつ自身がスピーカーの場合(CFP締め切り後)
+  # カンファレンス開催中、かつ自身がスピーカーの場合
+  # カンファレンス開催後、かつ自身がスピーカーの場合（開催後に資料URLを追加することを想定）
+  def display_speaker_dashboard_link?
+    (@conference.registered? && @conference.speaker_entry_enabled?) || (@conference.registered? && @speaker.present?) || (@conference.opened? && @speaker.present?) || (@conference.closed? && @speaker.present?)
```

Comment:
これでも動くとは思うんだけど

```suggestion
    @conference.speaker_entry_enabled? || @speaker.present?
```

でも行ける？

Reply:
conferenceがarchivedの時は非表示にしたいので、これだと表示されちゃうね。

これなら条件満たすとは思う
```
!@conference.archived? && (@conference.speaker_entry_enabled? || @speaker.present?) 
```

Reply:
表示を許可するstatusは明示的にしたいので今のままでいこう。

Reply:
OK